src/libtiled/hexagonalrenderer.cpp

```diff
@@ -67,23 +67,52 @@ QRect HexagonalRenderer::mapBoundingRect() const
 {
     const RenderParams p(map());
 
+    QRect mapBounds;
+
+    LayerIterator iterator(map());
+    while (Layer *layer = iterator.next()) {
+        if (TileLayer *tileLayer = dynamic_cast<TileLayer*>(layer))
+            mapBounds = mapBounds.united(tileLayer->bounds());
+    }
+
+    if (mapBounds.size() == QSize(0, 0))
+        mapBounds.setSize(QSize(1, 1));
+
     // The map size is the same regardless of which indexes are shifted.
     if (p.staggerX) {
-        QSize size(map()->width() * p.columnWidth + p.sideOffsetX,
-                   map()->height() * (p.tileHeight + p.sideLengthY));
+        if (!map()->infinite()) {
+            QSize size(map()->width() * p.columnWidth + p.sideOffsetX,
+                       map()->height() * (p.tileHeight + p.sideLengthY));
+
+            if (map()->width() > 1)
+                size.rheight() += p.rowHeight;
```

Comment:
I believe this line is also needed for infinite maps. `rowHeight` needs to be added because of the staggering. Same for `columnWidth` below.